[REF] commown_shareholder_register: various refactoring.

legalsylvain opened the pull request:
1. remove dependency to 'non standard' and non OCA module 'base_company_extension', present in Akretion repository.

In fact, this dependency is not required. It is set just to put in a form, the nominal_share_amount, just after the capital_amount. We now inherit from the base module, and but the nominal_share_amount just before the capital_amount.

Note: In fact, there is not direct relation between capital_amount, that is an administrative information, and the nominal share amount, that can change each day

legalsylvain committed:
[REF] commown_shareholder_register: remove dependency to 'non standard' and non OCA module 'base_company_extension', present in Akretion repository.
In fact, this dependency is not required. It is set just to put in a form, the nominal_share_amount, just after the capital_amount.
We now inherit from the base module, and but the nominal_share_amount just before the capital_amount.

Note: In fact, there is not direct relation between capital_amount, that is an administrative information, and the nominal share amount, that can change each day

legalsylvain committed:
[REF] commown_shareholder_register: Create demo data for colleges

legalsylvain committed:
[REF] commown_shareholder_register: configure demo data for company

fcayre commented:
Hello @legalsylvain ,

merci beaucoup pour tes contributions, je suis preneur aussi de ton avis général sur le module.

Pour ce qui concerne tes contributions, j'ai des questions :
- données de démo vs. créées dans le code : j'ai été récemment tenté de faire ça sur toute ma base de code, et j'ai lu je ne sais où que dans les versions récentes de Odoo (v18 ?) Odoo SA faisait le mouvement inverse, alors je me suis dit que je prendrai le temps plus tard de lire à ce sujet et j'ai retardé ce travail ; tu as lu ça qq part aussi ? un lien vers leurs arguments ? un avis de ton côté ?
- pour ce qui concerne l'insertion d'une CoA je suis bien embêté aussi, je trouve ça moche de faire ça dans tous nos modules, mais pour que les tests passent quand tous les modules sont installés, dont ceux qui dépendent de l18n_fr, j'ai pas trouvé d'autres solutions. Peut-être ne jamais dépendre de l18n_fr, à aucun niveau ? Comment tu fais toi ?

legalsylvain committed:
[FIX] commown_shareholder_register: access for two transient models. (sharehorlder_tags_update and register)

legalsylvain commented:
En fait, je suis pas au clair sur ce sujet.

Le déroulé de ma réflexion :  Si il y a des données de démo : 
-> quand on arrive devant un nouveau module qu'on connais pas, on peux immédiatement comprendre en partie comment l'utiliser. (ex : odoo / project : on voit comment ils modélisent un projet, des tâches, etc...)
-> ça augmente aussi le coverage naturellement. (tous les compute sont exécutés)
-> ça fait un jeu de démo persistent (au fur et à mesure des versions) pour faire des impressions d'écran à intégrer dans la doc.
-> on peut réutiliser ces données pour les tests, ce qui les rend plus léger.

Du coup, je trouve que c'est super, et je n'ai pas vu de direction chez odoo pour les virer. (mais je suis peut-etre passé à côté).

MAIS, dès que y'a de la compta, c'est la merde. Odoo est tout pété, selon que tu installes pas de CoA ou que y'a déjà un CoA installé, ça va tout merder. J'ai l'impression que la V18 va peut-etre un poil améliorer les choses, avec la suppression de l10n_generic_coa. (y'a [un CoA pocalypse par ici en V17](https://github.com/odoo/odoo/pull/110016), dont je connais pas tout les tenant et aboutissant). (> 1500 fichiers modifiés ;-))

Du coup, ma PR en cours pète la CI, ce qui est pas top.

J'ai donc fait un revert des changements liés à la compta.
- ça fait un diff plus simple
- la CI est contente.

Après, j'ai pas encore tout compris à ton module.



Autre point, j'ai une erreur d'absence de paramétrage. ça me permettait pas d'accéder  à l'édition du registre. j'ai donc rajouté un droit. J'imagine que ça a été fait manuellement sur la db, sinon, ça ne marcherait pas en prod' en V16 pour toi.

```
2025-06-30 03:01:43,838 5570 WARNING commown_shareholder_register_16_chaud odoo.modules.loading: The models ['commown_shareholder_register.register', 'commown_shareholder_register.shareholder_tags_update'] have no access rules in module commown_shareholder_register, consider adding some, like:
id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
commown_shareholder_register.access_commown_shareholder_register_register,access_commown_shareholder_register_register,commown_shareholder_register.model_commown_shareholder_register_register,base.group_user,1,0,0,0
commown_shareholder_register.access_commown_shareholder_register_shareholder_tags_update,access_commown_shareholder_register_shareholder_tags_update,commown_shareholder_register.model_commown_shareholder_register_shareholder_tags_update,base.group_user,1,0,0,0 
```

legalsylvain committed:
[REM] commown_shareholder_register: no need to have pot file. having it just add risk to have obsolete files. pot files are usefull only when:
Bot is updating pot file on each merge
a weblate is plugged, to update translation for each language

legalsylvain committed:
[FIX] commown_shareholder_register: entry menu name and update french translation

legalsylvain commented:
Review : 

- si le champs "rank" sert juste à ordonner les collèges, alors je pense qu'il faudrait le renommer "sequence", pour faire comme dans tout le reste de Odoo. (mais j'ai peut-etre pas compris le concept de ce champ).

- Je comprend pas le concept de collège et de catégorie. c'est vraiment un M2M ? j'aurais dit un O2M. si on peut en causer 5 minutes, ça me dit bien.
Ref : https://www.les-scic.coop/foire-aux-questions / https://www.l-expert-comptable.com/a/6074-societe-cooperative-d-interet-collectif-scic-definition-avantages.html

- le rapport ODS est clair et bien fait. c'est chouette, j'en avais jamais vu édité par Odoo.

- Question Use case : quand qq souscrit, même si il n'a pas payé, tu le considère comme au capital (il est dans le report ODS) ? il a le droit de vote du coup ? je croyais qu'il devait être validé par l'AG.

legalsylvain committed:
[IMP] commown_shareholder_register: improve tree view, using M2M tags widget

legalsylvain commented:
Après lecture, j'ai vraiment l'impression qu'on a un lien O2M de college qui peut contenir 1 à N catégories. Qu'en penses-tu ?

fcayre commented:
@legalsylvain je relis seulement maintenant les questions que tu soulèves ci-dessus, je ne sais plus si on a parlé de tout ça à l'oral donc je l'écris, tant pis si c'est une redite et désolé pour la réponse ultra-tardive sinon :
- oui tu as raison [c'est bien une O2M](https://github.com/commown/commown-odoo-addons/blob/16.0/commown_shareholder_register/models/shareholder_college.py#L15), je ne sais pas où tu as vu une M2M ou alors on ne parle pas de la même relation (?)
- le champ `rank` a un rôle fonctionnel c'est pour ça qu'on ne l'a pas appelé simplement `sequence` ; de plus c'est assez figé dans le temps, ça n'a pas vocation à être modifié hyper facilement par drag and drop. L'idée c'est que tu peux appartenir à 2 catégories, par exemple chez nous Bénéficiaire (parce que tu as souscript une action à 20€ lors de l'achat d'un service de location) et Investisseur (parce que plus tard tu t'es décidé à acheter 100€ de parts sociales), mais que dans ce cas on impose une seule catégorie, celle de rang le plus élevé (ici Investisseurs) car elle donne accès à un collège qui a plus de poids dans le vote en AG.
- de notre côté on ne valide pas les souscriptions en AG, je pense que c'est une règle statutaire, pas légale ; en revanche quelqu'un qui n'a pas payé ne devrait pas être considéré comme sociétaire, il est bien possible qu'on est un bug ici à un instant donné...

Honeyxilia commented:
Concernant la suppression de la dépendance à `base_company_extension`, nous avons ouvert une branche de notre côté : https://github.com/commown/commown-odoo-addons/pull/576

A noter que l'on utilise certaines des méthodes utilitaires du module en question dans les rapports ODS, et qu'on avait décidé de repousser la suppression à cause du code supplémentaire que l'on devrait rentrer chez nous, mais on priorisera en interne je pense
